include/nn/ui2d/Pane.h

```diff
@@ -5,8 +5,8 @@
 
 #pragma once
 
-#include "sead/runtime.h"  //FIXME this definitely is wrong
-#include "types.h"
+#include <nn/types.h>
+#include <prim/seadRuntimeTypeInfo.h>
```

Comment:
This sead include should be removed as it makes no sense for ui2d to be using sead

Reply:
I agree, but in various places, `nn` seems to use `getRuntimeTypeInfo`. If it cannot use the implementation by `sead`, does it have its own implementation? Is the real implementation maybe not within `sead`, but in `nn` in the first place?

Reply:
> nn seems to use getRuntimeTypeInfo

Yes and no. nn::pia uses pead, which is a fork of sead that is used for pia. nn::ui2d does *not* use sead or `getRuntimeTypeInfo`. Its RTTI function is called **G**etRuntimeTypeInfo (note the uppercase G).

> If it cannot use the implementation by sead, does it have its own implementation? 

Yes. Its 'own' implementation might turn out to be a copy-and-paste (or derivative) version of sead's, though.

> Is the real implementation maybe not within sead, but in nn in the first place?

No.

Reply:
seems to use a custom macro 
(the runtime info type is defined in `nn/font/font_Util.h`)
```cpp
namespace nn::font::detail {

class RuntimeTypeInfo {
  public:
    const RuntimeTypeInfo* m_ParentTypeInfo;

    explicit RuntimeTypeInfo(const RuntimeTypeInfo* parent) : m_ParentTypeInfo(parent) {}
    bool IsDerivedFrom(const RuntimeTypeInfo*) const;
};

}

// todo: figure out where to put this
#define NN_RUNTIME_TYPEINFO() 														\
	/* todo: implement this */														\
    static const nn::font::detail::RuntimeTypeInfo* GetRuntimeTypeInfoStatic(); 	\
																					\
    virtual const nn::font::detail::RuntimeTypeInfo* GetRuntimeTypeInfo() const { 	\
		static const nn::font::detail::RuntimeTypeInfo s_TypeInfo(nullptr); 		\
		return &s_TypeInfo; 														\
	}
```